README.md
# Payeezy bindings for WebBased app

Sample code provide Payeezy Javascript Integration

![alt tag](https://github.com/payeezy/payeezy_js/raw/master/example/PayeezyJS_GetToken.png)


1. Payeezy.js (which is included on the page using a &lt; script> tag) intercepts the form submit, and then
asynchronously posts the credit card details to Payeezy. This call uses JSONP over https, passing in API key and payment card details. 
2. On success Payeezy returns a token (type=FDtoken) to payeezy.js. Failure messages are returned in the response,
which can be handled as appropriate.
3. Payeezy.js then submits the form with the primary and secondary transactions as shown in the diagram above.

For more details on 'Payeezy JS' example refer [payeezy_js example](../../tree/master/example)

For more details on 'Payeezy JS Guide' refer [payeezy_js guide](../../blob/master/guide/payeezy_js062015.pdf)

# Token Based Transactions (NEW) - 
*	Generate Token with ta_token - auth false - GET API
*	Generate Token with ta_token - auth true - GET API
*	Generate Token without  ta_token & auth -  - GET API with 0$ Auth
*	Generate Token - Backward compatible -  GET API call

For more information, visit payeezy js guide

# Getting Started with Payeezy
Using below listed steps, you can easily integrate your mobile/web payment application with Payeezy APIs and go LIVE!
*	LITE  - REGISTRATION  
*	GET CERTIFIED
*	ADD MERCHANTS 
*	LIVE!

![alt tag](https://github.com/payeezy/get_started_with_payeezy/raw/master/payeezy_flow_diagram.png)

For more information on getting started, visit  [get_start_with_payeezy guide] (https://github.com/payeezy/get_started_with_payeezy/blob/master/get_started_with_payeezy042015.pdf) or [download](https://github.com/payeezy/get_started_with_payeezy/raw/master/get_started_with_payeezy042015.pdf)

# Testing - Payeezy {SANDBOX region}
For test credit card,eCheck,GiftCard please refer to [test data ](https://github.com/payeezy/testing_payeezy/blob/master/payeezy_testdata042015.pdf) or [download] (https://github.com/payeezy/testing_payeezy/raw/master/payeezy_testdata042015.pdf)

# Error code/response - Payeezy {SANDBOX/PROD region}
For HTTP status code, API Error codes and error description please refer to [API error code ](https://developer.payeezy.com/payeezy_new_docs/apis) and select your API

#Questions?
We're always happy to help with code or other questions you might have! Check out [FAQ page](https://developer.payeezy.com/faq-page) or call us. 

## Contributing

1. Fork it 
2. Create your feature branch (`git checkout -b my-new-feature`)
3. Commit your changes (`git commit -am 'Add some feature'`)
4. Push to the branch (`git push origin my-new-feature`)
5. Create a new Pull Request  

## Feedback

Payeezy JS is in active development. We appreciate the time you take to try it out and welcome your feedback!
Here are a few ways to get in touch:
* [GitHub Issues](https://github.com/payeezy/payeezy/issues) - For generally applicable issues and feedback
* support@payeezy.com - for personal support at any phase of integration
* [1.855.799.0790](tel:+18557990790)  - for personal support in real time 

## Terms of Use

Terms and conditions for using Payeezy JavaScript: Please see [Payeezy Terms & conditions](https://developer.payeezy.com/terms-use)
 
### License
The Payeezy JavaScript is open source and available under the MIT license. See the LICENSE file for more info.


CHANGELOG
=== 2.0.0 2015-06-20

Support tokenize method for credit card transactions.

# TokenBased Transactions (NEW) - 
*	Generate Token with ta_token - auth false - GET API
*	Generate Token with ta_token - auth true - GET API
*	Generate Token without  ta_token & auth -  - GET API with 0$ Auth
*	Generate Token - Backward compatible -  GET API call

=== 1.0.0 2014-09-01

* Initial release

Special thanks to: Sachin



javascript/payeezy.js
var Payeezy = function() {
    function e(e) {
        var t = {
            visa: /^4[0-9]{12}(?:[0-9]{3})?$/,
            mastercard: /^5[1-5][0-9]{14}$/,
            amex: /^3[47][0-9]{13}$/,
            diners: /^3(?:0[0-5]|[68][0-9])[0-9]{11}$/,
            discover: /^6(?:011|5[0-9]{2})[0-9]{12}$/,
            jcb: /^(?:2131|1800|35\d{3})\d{11}$/
        };
        for (var n in t) {
            if (t[n].test(e)) {
                return n
            }
        }
    }

    function t() {
        var e = [];
        var t = document.getElementsByTagName("input");
        var n = document.getElementsByTagName("select");
        for (i = 0; i < t.length; i++) {
            var r = t[i];
            var s = r.getAttribute("payeezy-data");
            if (s) {
                e[s] = r.value
            }
        }
        for (i = 0; i < n.length; i++) {
            var r = n[i];
            var s = r.getAttribute("payeezy-data");
            if (s) {
                e[s] = r.value
            }
        }
        return e
    }
    return {
        createToken: function(e) {
            var n = "api-cert.payeezy.com";
            this["clientCallback"] = e;
            var r = t();
            var i = 0;
            var s = {};
            var o = 0;
            var u = [];
            if (!this.apikey) {
                i = 400;
                u[o] = {
                    description: "Please set the API Key"
                };
                o++
            }
            if (!this.mercId) {
                i = 400;
                u[o] = {
                    description: "Please set the Merchant Identifier"
                }
            }
            if (u.length > 0) {
                s["error"] = {
                    messages: u
                };
                e(i, s);
                return false
            }
            var a = "https://" + n + "/v1/securitytokens?apikey=" + this.apikey + "&trtoken=" + this.mercId + "&callback=Payeezy.callback&type=transarmor&credit_card.type=" + r["card_type"] + "&credit_card.cardholder_name=" + r["cardholder_name"] + "&credit_card.card_number=" + r["cc_number"] + "&credit_card.exp_date=" + r["exp_month"] + r["exp_year"] + "&credit_card.cvv=" + r["cvv_code"];
            var f = document.createElement("script");
            f.src = a;
            document.getElementsByTagName("head")[0].appendChild(f)
        },
        setApiKey: function(e) {
            this["apikey"] = e
        },
        setMerchantIdentifier: function(e) {
            this["mercId"] = e
        },
        callback: function(e) {
            this["clientCallback"](e.status, e.results)
        }
    }
}();


javascript/payeezy_v2.js
var Payeezy = (function (){
	
		function getCardType(number) {
	        var cards = {
	            visa: /^4[0-9]{12}(?:[0-9]{3})?$/,
	            mastercard: /^5[1-5][0-9]{14}$/,
	            amex: /^3[47][0-9]{13}$/,
	            diners: /^3(?:0[0-5]|[68][0-9])[0-9]{11}$/,
	            discover: /^6(?:011|5[0-9]{2})[0-9]{12}$/,
	            jcb: /^(?:2131|1800|35\d{3})\d{11}$/
	        };
	        for (var card in cards) {
	            if (cards[card].test(number)) {
	                return card;
	            }
	        }
	    }
	
	function getFormData() {
		var formData = [];
		var inputElements = document.getElementsByTagName('input');
		var selectElements = document.getElementsByTagName('select');
		for(i=0; i<inputElements.length;i++) {
			var element = inputElements[i];
			var name = element.getAttribute('payeezy-data') 
			if (name) {
				formData[name] = element.value;
			}
		}
		for(i=0; i<selectElements.length;i++) {
			var element = selectElements[i];
			var name = element.getAttribute('payeezy-data') 
			if (name) {
				formData[name] = element.value;
			}
		}
		return formData;
	}
	

	return {
		createToken : function(callback) {
			var host = 'api-cert.payeezy.com';
			this['clientCallback'] = callback;
			var formData = getFormData();
			var status =0;
			var result = {};
			var i=0;
			var messageArray = [];
			if (!this.apikey) {
				status = 400;
				messageArray[i] = {description : 'Please set the API Key'};
				i++;
			}
			if (!this.mercId) {
				status = 400;
				messageArray[i] = {description : 'Please set the Merchant Identifier'};
			}
			
			if (messageArray.length > 0) {
				result['error'] = {messages : messageArray};
				callback(status, result);
				return false;
			}
			var endpoint = 'https://'+host+'/v1/securitytokens?apikey='+this.apikey+'&trtoken='+this.mercId+'&callback=Payeezy.callback&type=FDToken&auth='+formData['auth']+'&ta_token='+formData['ta_token']+'&credit_card.type='+formData['card_type']+'&credit_card.cardholder_name='+formData['cardholder_name']+'&credit_card.card_number='+formData['cc_number']+'&credit_card.exp_date='+formData['exp_month']+formData['exp_year']+'&credit_card.cvv='+formData['cvv_code'];
			
			var scriptTag = document.createElement('script');
			scriptTag.src = endpoint;
			
			document.getElementsByTagName('head')[0].appendChild(scriptTag);
		},
		setApiKey : function(apiKey) {
			this['apikey'] = apiKey;
		},
		setMerchantIdentifier : function(mercId) {
			this['mercId'] = mercId;
		},
		callback : function (response) {
			this['clientCallback'](response.status, response.results);
		}
	};
})();

